src/content/reference/react/useTransition.md

```diff
@@ -1501,6 +1502,92 @@ main {
 
 ---
 
+### Displaying an error to users with a error boundary {/*displaying-an-error-to-users-with-error-boundary*/}
+
+<Canary>
+
+Error Boundary for useTransition is currently only available in React's canary and experimental channels. Learn more about [React's release channels here](/community/versioning-policy#all-release-channels).
+
+</Canary>
+
+If you'd like to display an error to your users when using `useTransition`, you can use an [error boundary](/reference/react/Component#catching-rendering-errors-with-an-error-boundary). To use an error boundary, wrap the component where you are calling the `useTransition` in an error boundary. If the function in `startTransition` is rejected the fallback for the error boundary will be displayed.
+
+<Sandpack>
+
+```js SubmitContainer.js active
+
+import { useTransition } from "react";
+import { ErrorBoundary } from "react-error-boundary";
+
+export function SubmitContainer() {
+  return (
+    <ErrorBoundary fallback={<p>⚠️Something went wrong</p>}>
+        <SubmitButton />
+    </ErrorBoundary>
+  );
+}
+
+function SubmitButton() {
+  const [pending, startTransition] = useTransition();
+
+  return (
+    <button 
+      disabled={pending} 
+      onClick={() => {
+        startTransition(() => {
+          throw new Error();
+        });
+      }}>
+        Submit
```

Comment:
Maybe we could make this button label be like: "Trigger Error Boundary" to be clearer

Reply:
Or, lean into a real-world example where we'd use a `useTransition` like the example of loading indicators: https://react-nzxyyeyvn-fbopensource.vercel.app/reference/react/useTransition#preventing-unwanted-loading-indicators

Reply:
I'll update this example to be more product specific based on yours and @sophiebits's suggestions